Tests/ClientRuntimeTests/NetworkingTests/FlexibleChecksumsMiddlewareTests.swift

```diff
@@ -0,0 +1,405 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import AwsCommonRuntimeKit
+import SmithyTestUtil
+@testable import ClientRuntime
+
+class FlexibleChecksumsMiddlewareTests: XCTestCase {
```

Comment:
I reordered some of the steps in these tests since the middleware doesn't call `next()` at the very end of execution anymore.